packages/babel-core/src/config/config-chain.js

```diff
@@ -419,7 +546,7 @@ function matchesPatterns(
 
     const absolutePatterns = strings.map(pattern => {
       // Preserve the "!" prefix so that micromatch can use it for negation.
-      const negate = pattern[0] === "!";
+      const negate = pattern[0] === "!" && !disableNegation;
```

Comment:
I think we should throw an error/report a warning if `disableNegate && pattern[0] === "!"`

Reply:
`!disableNegation` is not super clear to me but I don't have a better idea.

Reply:
Good call, happy to throw instead.